### ISSUE TITLE
Path substition for passwords.xml on network share not working

### ISSUE THREAD

**Issue by ultraviolet7** [xbmc/xbmc#16358]
### Describe the bug
Here is a clear and concise description of what the problem is:

Path substitution in general appears to be mostly fixed thanks to #15971 but using advancedsettings.xml for path substitution of passwords.xml located on a network share causes a crash at startup.  The crash occurs even when passwords.xml contains no passwords.
````
<passwords>
</passwords>
````
Path substitution of passwords.xml within the local file system, as opposed to a network location, appears to work correctly.  The network location of my passwords.xml file does _**not**_ itself require a password as it is using NFS.  This previously worked (did not crash) in v17.6.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Path substitution of passwords.xml located on a network share should work like any other path substitution.

## Actual Behavior

Kodi crashes at startup when path substitution of passwords.xml points to a network share.

### To Reproduce

1. Add to your advancedsettings.xml the following:
````
<substitute>
  <from>special://profile/passwords.xml</from>
  <to>nfs://YOUR_SERVER/YOUR_PATH/userdata/passwords.xml</to>
</substitute>
````
2. Launch Kodi

### Debuglog

The debuglog can be found here:
https://paste.kodi.tv/faripaxehu.kodi

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [x] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: macOS 10.13.6
 - Kodi version: 18.3

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by DaVukovic** [comment]
You are susbtituting pretty much, which is not recommended due to our Wiki anyway: 

> Path substitution is not officially supported for settings files.

Are you sure it's the password.xml substitute which gives you the error?

Have you tried with an NFS share which was mounted by the OS before and you simply use the local mountpoint for the `<to>` section rather than a full network path?

**Comment by ksooo** [comment]
I guess Kodi tries to lookup the password for the share the passwords.xml is located on. 

Ofc Kodi should not crash in this scenario. But obviously password protecting the share passwords.xml is located on weil never work.

**Comment by ultraviolet7** [comment]
@DaVukovic 
Thank you for the note from the Wiki.  I missed that in my research into this problem.  The section of the Wiki talking about using MySQL shows an example of using path substitution for the passwords.xml file.  https://kodi.wiki/view/MySQL/Sync_other_parts_of_Kodi#Network_share_passwords
Yes, it is the passwords.xml substitution that causes the error.  I narrowed it down by commenting out the substitutions one by one until I found that the passwords.xml was causing the problem.
I went ahead and tried your suggestion about mounting the the NFS share in the OS and modifying my advancedsettings.xml to use that path instead.  This does work as Kodi now sees this as a local file.  This is probably a usable workaround for my macOS clients but will probably not work with my Android clients (I have not tired yet).

@ksooo 
I agree that in the scenario where a password is needed to access the share where passwords.xml is located is not going to work.  In my situation a password is not needed to access the share as I am using NFS.
